### PR TITLE
custom port option

### PR DESCRIPTION
**Pull request by subtosharki**
## Bug

- [ ] Related issues linked using `closes: #number`

## Feature

- [x] Related issues linked using `closes: #number`
- [x] There is only 1 db migration with no breaking changes.

## Translations

- [ ] `.json` files are formatted: `yarn format`
- [ ] Translations are correct
- [ ] New translation? It's been added to `next.config.js`

### CODE REVIEW

File: scripts/copy-env.mjs [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
 import { join } from "node:path";
+import "dotenv/config";
+import * as fs from "fs";
+
+fs.readFile("./packages/client/package.json", "utf8", (err, json) => {
+
+  let port = process.env.PORT_CLIENT;
+  if (!port) {
+    return;
+  } else {
```
Comment (Dev-CasperTheGhost) [inline scripts/copy-env.mjs:11]:
```suggestion
  if (port) {
```

Comment (subtosharki) [reply scripts/copy-env.mjs:11]: understoof

**Review by subtosharki — COMMENTED** [review]
looks fine

**Comment by subtosharki** [comment]
ready for merge ?

File: scripts/copy-env.mjs [hunk 2]
```diff
@@ -1,5 +1,22 @@
 import { one } from "copy";
 import { join } from "node:path";
+import "dotenv/config";
+import { readFileSync, writeFileSync } from "fs";
+
+fs.readFile("./packages/client/package.json", "utf8", (err, json) => {
```
Comment (Dev-CasperTheGhost) [inline scripts/copy-env.mjs:6]:
```suggestion
readFileSync("./packages/client/package.json", "utf8", (err, json) => {
```

File: scripts/copy-env.mjs [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+
+  const port = process.env.PORT_CLIENT;
+  if (port) {
+    json = JSON.parse(json);
+    json.scripts.start = `next start -p ${port}`;
+    json = JSON.stringify(json, null, 2);
+    fs.writeFile("./packages/client/package.json", json, (err) => {
```
Comment (Dev-CasperTheGhost) [inline scripts/copy-env.mjs:13]:
```suggestion
    writeFileSync("./packages/client/package.json", json, (err) => {
```

**Review by Dev-CasperTheGhost — CHANGES_REQUESTED** [review]
Just some minor fixes

**Comment by Dev-CasperTheGhost** [comment]
> ready for merge ?

Not yet. I’m going to fix so it can be run within the packages folder, later today

**Comment by subtosharki** [comment]
> Oh, are you able to convert the callbacks into non-callbacks? That should resolve the failing ci.
> 
> If not, I can do that when I get home

im at school rn

**Comment by Dev-CasperTheGhost** [comment]
> > Oh, are you able to convert the callbacks into non-callbacks? That should resolve the failing ci.
> > If not, I can do that when I get home
> 
> im at school rn

Yeah, same hahha! I’ll do this when I’m home.

**Review by Dev-CasperTheGhost — APPROVED** [review]
Looks good now!

**Pull request merged**